src/components/ShareStructure/EditShareStructure.vue

```diff
@@ -280,7 +280,10 @@ export default class EditShareStructure extends Mixins(CurrencyLookupMixin) {
     let rules: Array<Function> = []
     if (!this.hasNoMaximumShares) {
       rules = [
-        (v: string) => /^\d+$/.test(v) || 'Must be a number greater than 0']
+        (v: string) => (v !== null && v !== '') || 'Number of shares is required',
+        (v: string) => /^-?\d+$/.test(v) || 'Must be a whole number',
```

Comment:
I allow for an optional "-" here so this rule succeeds and the next rule catches it. (better ux)